### PR TITLE
add name for region settings

### PR DESCRIPTION
**Pull request by Samweli**
Fixes https://github.com/ConservationInternational/trends.earth/issues/368

Screenshot
settings_name

### CODE REVIEW

**Comment by luipir** [comment]
ok, I'll give a look as soon as possible @Samweli

**Comment by luipir** [comment]
@Samweli just a quick though seeing the animated gif... if the name have to be used as folder or file name, what else to avoid to use "(" or other "." other that used in file extension?

File: LDMP/settings.py [hunk 1]
```diff
@@ -235,6 +238,15 @@ def __init__(self, parent=None):
         self.populate_cities()
         self.area_admin_0.currentIndexChanged.connect(self.populate_admin_1)
         self.area_admin_0.currentIndexChanged.connect(self.populate_cities)
+        self.area_admin_0.currentIndexChanged.connect(self.generate_name_setting)
+
+        self.secondLevel_area_admin_1.currentIndexChanged.connect(self.generate_name_setting)
```
Comment (luipir) [inline LDMP/settings.py:243]: what would be the meaning of "admin_1" in secondLevel_area_admin_1 combobox menu?

File: LDMP/settings.py [hunk 2]
```diff
@@ -301,6 +313,9 @@ def load_settings(self):
         if buffer_size:
             self.buffer_size_km.setValue(float(buffer_size))
         self.checkbox_buffer.setChecked(buffer_checked)
+        self.area_settings_name.setText(
+            self.settings.value("trends_earth/region_of_interest/area_settings_name", '')
```
Comment (luipir) [inline LDMP/settings.py:317]: as far as I can see  there is a principal problem in this code is that if a use set a custom name, any area parameter modification will override custom string.
Would it make sense to add a checkbox locking the generated or custom string?

**Comment by luipir** [comment]
@Samweli other the aspect to lock the area name, IMHO LGTM. Need to have opinion by @ricardogsilva about a locking checkbox and eventually removing some special character in ara name

**Comment by ricardogsilva** [comment]
I think it is OK to let the code regenerate the custom name whenever the user changes some area parameter for now.

**Comment by ricardogsilva** [comment]
@luipir 

if you are :+1: on this PR, please proceed to merge

File: LDMP/gui/WidgetSelectArea.ui [hunk 3]
```diff
@@ -522,6 +507,20 @@
         </layout>
        </widget>
       </item>
+      <item>
+       <layout class="QHBoxLayout" name="horizontalLayout_2">
+        <item>
+         <widget class="QLabel" name="label_2">
+          <property name="text">
+           <string>Settings name:</string>
```
Comment (luipir) [inline LDMP/gui/WidgetSelectArea.ui:515]:
```suggestion
           <string>Area name:</string>
```

I'm confused about the label you used for the tag of the Area.
Wait for resolve this to merge

**Comment by luipir** [comment]
@Samweli would it make sense to set the first time value in no setting is available calling one time generate_name_setting() ?
I started the setting GUI and area name is set to empty... is the desired behaviour?

**Comment by luipir** [comment]
just as note and out of the scope of the PR... the area name continue to have the same localization problem as the region/city giving back a strange string.

**Review by luipir — CHANGES_REQUESTED** [review]

**Comment by Samweli** [comment]
> @Samweli just a quick though seeing the animated gif... if the name have to be used as folder or file name, what else to avoid to use "(" or other "." other that used in file extension?

@luipir we are not saving the area name as a file name or folder, so I guess we won't have to do a check for those characters.

File: LDMP/settings.py (hunk 1 above)
Comment (Samweli) [reply LDMP/settings.py:243]: I'm just reusing the widgets name as defined from the area widget. Do you want us to update them?

File: LDMP/settings.py (hunk 2 above)
Comment (Samweli) [reply LDMP/settings.py:317]: @luipir I think that is the expected behaviour, also this is only executed once when opening up the settings dialog.

**Comment by Samweli** [comment]
> @Samweli would it make sense to set the first time value in no setting is available calling one time generate_name_setting() ?
> I started the setting GUI and area name is set to empty... is the desired behaviour?

Yes I think that right @luipir, we should not set it to empty when using the dialog for the first time, @ricardogsilva what do you think?

File: LDMP/gui/WidgetSelectArea.ui (hunk 3 above)
Comment (Samweli) [reply LDMP/gui/WidgetSelectArea.ui:515]: ok let me change it.

File: LDMP/settings.py (hunk 1 above)
Comment (luipir) [reply LDMP/settings.py:243]: no... it's ok

File: LDMP/settings.py (hunk 2 above)
Comment (luipir) [reply LDMP/settings.py:317]: ok

**Review by luipir — APPROVED** [review]

**Comment by luipir** [comment]
> Yes I think that right @luipir, we should not set it to empty when using the dialog for the first time, @ricardogsilva what do you think?

just to speedup I merge it, we can manage detail (like first name value setup) in an issue later @Samweli

**Comment by Samweli** [comment]
thanks @luipir

**Pull request merged**